content/ja/handbook/engineering/architecture/design-documents/gitlab_messaging_layer/index.md

```
@@ -0,0 +1,588 @@
+<td class="px-3 py-2 border border-gray-300">2025-02-26</td>
+</tr>
+</tbody>
+</table>
+</div>
+
+
+## 概要
+
+このドキュメントでは、GitLab スケールでのデータメッセージングとキューイングのニーズをサポートするための _メッセージングレイヤー_ の採用・使用に関する設計、アーキテクチャ、およびロールアウトロードマップを提案します。
+
+[Data Insights Platform](https://docs.google.com/document/d/1V3XRXfPquBrI_-ob9Fn2Jdskq7W4-heG6zBjJ66AOx8/edit?usp=sharing) の構築や [Project Siphon](/handbook/engineering/architecture/design-documents/siphon/) などの最近のイニシアチブから、大量のデータを取り込み・処理するためにテクノロジースタック内にスケーラブルで信頼性の高いキューイングシステムが必要であることが明らかになっています。[この点に関する複数の議論](#additional-context) を経て、選択肢を [NATS の使用](https://docs.nats.io/nats-concepts/what-is-nats) に絞り込みました。その理由はこのドキュメントの後半で説明します。
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)